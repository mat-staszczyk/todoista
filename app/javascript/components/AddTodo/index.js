import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { Container, Input, Button } from 'semantic-ui-react';
import { addTodo } from '../../actions';

const AddTodo = ({ dispatch }) => {
  let input;

  return (
    <Container>      
      <Input
        ref={node => { input = node ? node.inputRef : null }}
      />
      <Button
        basic 
        color='teal'
        onClick={() => {
          dispatch(addTodo(input.value));
          input.value = '';
        }}
      >
        Add Todo
      </Button>
    </Container>
  );
};

AddTodo.propTypes = {
  dispatch: PropTypes.func.isRequired,
};

export default connect()(AddTodo);