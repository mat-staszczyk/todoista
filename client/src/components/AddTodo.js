import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { Container, Input, Button } from 'semantic-ui-react';
import { addTodo } from '../actions';

const AddTodo = ({ dispatch }) => {
  let input;

  return (
    <Container>      
      <Input
        ref={node => { input = node.inputRef }}
      />
      <Button
        basic 
        color='teal'
        onClick={() => {
          console.log('value test', input.value);
          dispatch(addTodo(input.value));
          input.value = '';
        }}
      >
        Add Todo
      </Button>
    </Container>
  );
};

AddTodo.propTypes = {
  dispatch: PropTypes.func.isRequired,
};

export default connect()(AddTodo);