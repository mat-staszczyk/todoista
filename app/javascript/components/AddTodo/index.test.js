import React from 'react';
import AddTodo from '.';
import configureStore from 'redux-mock-store';
const mockStore = configureStore();
import renderer from 'react-test-renderer';

const createExampleProps = () => ({
  dispatch: jest.fn(),
  store: mockStore(),
});

describe('<AddTodo />', () => {
  it('renders properly', () => {
    const exampleProps = createExampleProps();

    expect(renderer.create(<AddTodo {...exampleProps} />).toJSON()).toMatchSnapshot();
  })
})
