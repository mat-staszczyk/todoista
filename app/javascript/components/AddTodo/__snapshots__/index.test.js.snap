// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTodo /> renders properly 1`] = `
<div
  className="ui container"
>
  <div
    className="ui input"
  >
    <input
      disabled={undefined}
      onChange={[Function]}
      tabIndex={undefined}
      type="text"
    />
  </div>
  <button
    className="ui teal basic button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
  >
    Add Todo
  </button>
</div>
`;
